components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/impl/ActionExecutorServiceImpl.java

```diff
@@ -156,18 +162,18 @@ private void validateActionIdList(ActionType actionType, String[] actionIdList)
         }
     }
 
-    private ActionExecutionStatus execute(Action action, Map<String, Object> eventContext)
+    private ActionExecutionStatus<?> execute(Action action, Map<String, Object> eventContext)
             throws ActionExecutionException {
 
         ActionType actionType = ActionType.valueOf(action.getType().getActionType());
         ActionExecutionRequest actionRequest = buildActionExecutionRequest(actionType, eventContext);
         ActionExecutionResponseProcessor actionExecutionResponseProcessor = getResponseProcessor(actionType);
 
-        return Optional.ofNullable(action)
-                .filter(activeAction -> activeAction.getStatus() == Action.Status.ACTIVE)
-                .map(activeAction -> executeAction(activeAction, actionRequest, eventContext,
-                        actionExecutionResponseProcessor))
-                .orElse(new ActionExecutionStatus(ActionExecutionStatus.Status.FAILED, eventContext));
+        if (action.getStatus() == Action.Status.ACTIVE) {
+            return executeAction(action, actionRequest, eventContext, actionExecutionResponseProcessor);
+        } else {
+            return new SuccessStatus.Builder().setResponseContext(eventContext).build();
```

Comment:
Let's add a comment here mentioning no actions to execute is considered as action execution is success

Reply:
This is fixed with 148073eab6305adc3e26c71ca755a0550e805f54